pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -437,6 +437,17 @@ func (s *NodeController) deletePods(nodeID string) error {
 		if err := s.kubeClient.Pods(pod.Namespace).Delete(pod.Name); err != nil {
 			glog.Errorf("Error deleting pod %v: %v", pod.Name, err)
 		}
+		if pod.Spec.RestartPolicy == api.RestartPolicyAlways {
```

Comment:
Some kind of pod controller, yes, if they want them to survive node failure, even if replicas is 1.

See:
https://github.com/GoogleCloudPlatform/kubernetes/pull/5353/files